flipperdevices/flipperzero-firmware
# [FL-3179] 1-Wire Overdrive Mode

# What's new

- Add overdrive mode support for `one_wire` library.
- Implement overdrive reading, writing and emulation for DS1996.
- Update the `example_thermo` demo application.

# Verification 

- Check all 1-Wire and iButton functionality for regressions. All existing functionality must remain intact.
- Unzip [Test DS1996.ibtn.zip](https://github.com/flipperdevices/flipperzero-firmware/files/11027716/Test.DS1996.ibtn.zip) and copy the file to `/ext/ibutton`.
- Go to `iButton -> Saved -> Test DS1996 -> Emulate` and launch the emulation.
- Read the emulated iButton with another Flipper or with a reader such as iKey. The data should be read fully and correctly. In the first case, both development and PR versions should be able to read it, but the latter should be several times faster.
- (Optional) Go to `iButton -> Read` and read a real DS1996 iButton.
- (Optional) `iButton -> Saved -> Test DS1996 -> Write Copy` and copy the file onto a real DS1996. Read it back afterwards to ensure that the data was written correctly.

# Checklist (For Reviewer)
